test/fft_test.rg

```diff
@@ -17,92 +17,415 @@ import "regent"
 local fft = require("fft")
 
 local cmapper = require("test_mapper")
+local format = require("std/format")
 
-local fft1d = fft.generate_fft_interface(int1d, complex64)
-local fft2d = fft.generate_fft_interface(int2d, complex64)
+-- PRINT FUNCTIONS
+
+local function make_print_region_task(title, input)
+  local t = regentlib.newsymbol(title, "t")
+  local i = regentlib.newsymbol(input, "i")
+  local task print_region_task([t], [i])
+  where reads (i) do
+    format.println("{} = [", t)
+    for x in i do
+      format.println("{},", i[x])
+    end
+    format.println("]")
+  end
+  return print_region_task
+end
+
+local print_region_1d_float = make_print_region_task(rawstring, region(ispace(int1d), float))
+local print_region_1d_double = make_print_region_task(rawstring, region(ispace(int1d), double))
+local print_region_1d_complex32 = make_print_region_task(rawstring, region(ispace(int1d), complex32))
+local print_region_1d_complex64 = make_print_region_task(rawstring, region(ispace(int1d), complex64))
+local print_region_2d_complex64 = make_print_region_task(rawstring, region(ispace(int2d), complex64))
+local print_region_3d_double = make_print_region_task(rawstring, region(ispace(int3d), double))
+local print_region_3d_complex64 = make_print_region_task(rawstring, region(ispace(int3d), complex64))
+local print_region_4d_complex64 = make_print_region_task(rawstring, region(ispace(int4d), complex64))
+
+-- COMPARISON FUNCTIONS
+
+local function make_compare_regions_task(output, expected)
+  local o = regentlib.newsymbol(output, "o")
+  local e = regentlib.newsymbol(expected, "e")
+  local task compare_region_task([o], [e])
+  where reads (o, e) do
+    var status = "PASSED"
```

Comment:
Why use a string here? It seems more reasonable to calculate and return a boolean and manage how we print the result to the user on the caller's side. Plus, I'm 99.9% sure this will crash on multiple nodes.

Reply:
Added a boolean and asserts: https://github.com/arjunkunna/regent-fft/commit/686a130f6f12c8154a24511b58248d6a22506470